cloudinit/sources/DataSourceEc2.py

```diff
@@ -234,7 +234,7 @@ def wait_for_metadata_service(self):
 
         # try the api token path first
         metadata_address = self._maybe_fetch_api_token(mdurls)
-        if not metadata_address:
+        if not metadata_address and self.cloud_name != CloudNames.AWS:
```

Comment:
This seems too broad a skip on CloudNames.AWS platform as we will now not fall back to IMDSv1 at all if we are AWS and there was a timeout on the IMDSv2 PUT for some reason.  No retries would be tried in this case and I believe the logic would take us down the path of not crawling any metadata in that case, also rejecting Ec2 as a datasource.
I really think we need to specifically check exc.code == 403 and ! AWS automatically fallback to IMDSv1 in that case.